File: Remove target Element/Remove target element from array.py
"""
Problem
--------
Input: 
nums = [0,1,2,2,3,0,4,2], target = 2

Output: nums = [0,1,4,0,3]

Input: 
nums = [3,2,2,3], target = 3

Output: nums = [2,2]

-------------------------
| Prepared by -         |
| Himel Sarder          |
| Dept. of CSE, BSFMSTU |
-------------------------
"""
def remove(nums):
    if not nums:
        return 0

    i = 0

    while i < len(nums):
        if nums[i] == target:
            nums.pop(i)
            
        else:
            i += 1


# Example 1
nums1 = [0,1,2,2,3,0,4,2]
target = 2
k1 = remove(nums1)
print(nums1)  # Output: [0,1,4,0,3]

# Example 2
nums2 = [3,2,2,3]
target = 3
k2 = remove(nums2)
print(nums2)  # Output: [2,2]

File: Remove target Element/Remove target Element from array.cpp
#include <iostream>
#include <vector>

using namespace std;

void remove(vector<int>& nums, int target) {
    if (nums.empty()) {
        return;
    }

    int i = 0;

    while (i < nums.size()) {
        if (nums[i] == target) {
            nums.erase(nums.begin() + i);
        } else {
            i++;
        }
    }
}

int main() {
    // Example 1
    vector<int> nums1 = {0, 1, 2, 2, 3, 0, 4, 2};
    int target1 = 2;
    remove(nums1, target1);

    cout << "["; // Start with an opening square bracket

    for (int i = 0; i < nums1.size(); i++) {
        cout << nums1[i]; // Print the current number

        if (i < nums1.size() - 1) {
            cout << ", "; // Add a comma and space if it's not the last element
        }
    }

    cout << "]" << endl; // End with a closing square bracket and newline

    // Example 2
    vector<int> nums2 = {3, 2, 2, 3};
    int target2 = 3;
    remove(nums2, target2);

    cout << "["; // Start with an opening square bracket

    for (int i = 0; i < nums2.size(); i++) {
        cout << nums2[i]; // Print the current number

        if (i < nums2.size() - 1) {
            cout << ", "; // Add a comma and space if it's not the last element
        }
    }

    cout << "]" << endl; // End with a closing square bracket and newline

    return 0;
}

-------------------------
| Prepared by -         |
| Himel Sarder          |
| Dept. of CSE, BSFMSTU |
-------------------------
